ddtrace/tracer/tracer.go

```diff
@@ -153,6 +153,24 @@ func Inject(ctx ddtrace.SpanContext, carrier interface{}) error {
 	return internal.GetGlobalTracer().Inject(ctx, carrier)
 }
 
+// SetUser tags the root span of s with the provided user id as well as the
+// optional tags provided as userTagOption parameters. This allows user identification
+// across all spans of a trace.
+func SetUser(s Span, id string, opts ...UserTagOption) {
+	if s == nil {
+		return
+	}
+	if span, ok := s.(*span); ok {
```

Comment:
You should probably have the else branch in order to still set the tag.

In the longer term, I think the APM team should introduce a more advance interface allowing to set the root spans. And this is a good enough solution in my opinion.